assets/src/api.ts

```diff
@@ -144,6 +154,32 @@ export const apiCallWithError = <T, U>({
       return fetchError()
     })
 
+export const apiCallResult = async <T, E>(
+  url: Parameters<typeof fetch>[0],
+  OkStruct: Struct<T, unknown>,
+  ErrStruct: Struct<E, unknown>,
+  requestInit?: Parameters<typeof fetch>[1]
+): Promise<Result<T, E>> =>
+  fetch(url, requestInit)
+    .then(async (response) => {
+      const json: unknown = await response.json()
+
+      if (response.ok && is(json, object({ data: any() }))) {
+        return Ok(create(json.data, OkStruct))
+      } else {
+        assert(json, object({ error: any() }))
+        return Err(create(json.error, ErrStruct))
+      }
+    })
+    .catch((error) => {
+      if (error instanceof StructError) {
+        Sentry.captureException(error)
+      }
+      // We throw within the returned promise because downstream consumers
+      // should handle errors themselves
+      throw error
+    })
+
```

Comment:
@mbta/skate-developers any thoughts on keeping this in `api.ts` vs it's own file?

If voting for moving it out, what would you name the file?

Reply:
Hmmm - I think it makes sense where it is, but I'm not _opposed_ to moving it?

HOWEVER, thinking about what file I would put this in made me start bikeshedding the name of the function. `apiCallResult` makes me think that it's transforming some data into a nicely formatted result, which it IS doing, but I would want the name to also indicate that it's actually doing the API call. I'm not sure what TO call it though - `fetchFromApi` or something like that?

This is non-blocking, obviously. We can rename it later if a name idea speaks to us.